crates/phactory/src/contracts/btc_price_bot.rs

```diff
@@ -0,0 +1,283 @@
+///
+/// For now, you cannot use `tokio`-based HTTP crate since it is not compatible in SGX. Read more about the details in our
+/// PR <https://github.com/Phala-Network/phala-blockchain/pull/483> for the reason why.
+///
+/// For more side task demos, visit <https://github.com/Phala-Network/phala-blockchain/tree/side-task-demo1> and
+/// <https://github.com/Phala-Network/phala-blockchain/tree/side-task-demo2>.
+///
+/// For the basic functionalities of contract, refer to `guess_number.rs`.
+
+#[derive(Encode, Decode, Debug, Clone, Default)]
+pub struct BtcPriceBot {
+    owner: AccountId,
+    bot_token: String,
+    chat_id: String,
+    price: String,
+}
+
+/// The Queries to this contract
+///
+/// End users query the contract state by directly sending Queries to the pRuntime without going on chain.
+/// They should not change the contract state.
+#[derive(Encode, Decode, Debug, Clone)]
+pub enum Request {
+    /// Query the current owner of the contract
+    QueryOwner,
+    /// Query the authentication token of telegram bot
+    /// refer to: https://core.telegram.org/bots/api#authorizing-your-bot
+    QueryBotToken,
+    /// Query the identifier to target chat
+    /// refer to: https://core.telegram.org/bots/api#sendmessage
+    QueryChatId,
+    /// Query the cached BTC price
+    QueryPrice,
+}
+
+/// The Query results
+#[derive(Encode, Decode, Debug, Clone)]
+pub enum Response {
+    Owner(AccountId),
+    BotToken(String),
+    ChatId(String),
+    Price(String),
+}
+
+#[derive(Encode, Decode, Debug, thiserror::Error)]
+#[error("{:?}", self)]
+pub enum Error {
+    OriginUnavailable,
+    NotAuthorized,
+    NetworkUnavailable,
+    BadBtcPrice,
+}
+
+impl BtcPriceBot {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+/// The payloads of the Telegram `sendMessage` request
+/// refer to: https://core.telegram.org/bots/api#sendmessage
+#[derive(Deserialize, Serialize)]
+struct TgMessage {
+    chat_id: String,
+    text: String,
+}
+
+/// The BTC price from https://min-api.cryptocompare.com
+#[derive(Deserialize, Serialize, Debug)]
+struct BtcPrice {
+    #[serde(rename(deserialize = "USD"))]
+    usd: f64,
+}
+
+// Alice is the pre-defined root account in dev mode
+const ALICE: &str = "d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d";
+
+impl contracts::NativeContract for BtcPriceBot {
+    type Cmd = Command;
+    type QReq = Request;
+    type QResp = Result<Response, Error>;
+
+    /// Return the contract id which uniquely identifies the contract
+    fn id(&self) -> contracts::ContractId {
+        contracts::id256(contracts::BTC_PRICE_BOT)
+    }
+
+    /// Handle the Commands from transactions on the blockchain. This method doesn't respond.
+    fn handle_command(
+        &mut self,
+        origin: MessageOrigin,
+        cmd: Command,
+        context: &mut NativeContext,
+    ) -> TransactionResult {
+        info!("Command received: {:?}", &cmd);
```

Comment:
Make it a comment. If we do this, the secrets in the command will be leaked to the miners

Reply:
done